### DISCUSSION TITLE
Works fine on windows, receive nothing on raspberry pi

### DISCUSSION THREAD

**Discussion by jjd7076** [jvde-github/AIS-catcher#346] [General]
I tested out AIS-catcher on windows and everything worked fine.  I was tracking several ships on the river I live next to.  I thought it was cool, so I set up a raspberry pi 4 to run stand alone.  I got everything up and running.. rtl is running, web interface working..  but I'm not receiving any messages on the pi.  I'm using the same RTL-sdr v4 and the same exact antenna.  I tried different tuner settings, sample rates, AGC on/off, yet nothing.  What could be wrong? Any help with this would be greatly appreciated. 

Thank You

**Comment by jvde-github** [comment]
How did you install? And did you install librtlsdr by hand?

**Reply by jjd7076** [reply]
I installed the rtl-sdr via terminal "sudo apt install rtl-sdr", then I followed the instructions in the installation guide.

**Reply by jvde-github** [reply]
The Debian packages do not have the latest RTLSDR library with V4. That is why my script installs it from source but not if it is already available. Can you try to install via: 
```
sudo bash -c "$(wget -qO- https://raw.githubusercontent.com/jvde-github/AIS-catcher/main/scripts/aiscatcher-install) _ -p"
```

And try to start it and if it does not work show the output? 

The executable is in `/usr/bin/AIS-catcher`

**Reply by jjd7076** [reply]
This is what I'm getting:

AIS-catcher (build Dec 24 2024) v0.61-124-ge6c7faae
(C) Copyright 2021-2025 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Adding plugin (/etc/AIS-catcher/plugins/shipxplorer_option.pjs). Description: "add option to look up vessel details in ShipXplorer", Author: "jvde-github", version 3
Adding plugin (/etc/AIS-catcher/plugins/marinetraffic_option.pjs). Description: "add option to look up vessel details in MarineTraffic", Author: "jvde-github", version 3
Adding plugin (/etc/AIS-catcher/plugins/google_option.pjs). Description: "add option to look up location in Google Maps", Author: "jvde-github", version 3
Adding plugin (/etc/AIS-catcher/plugins/vesselfinder_option.pjs). Description: "add option to look up vessel details in VesselFinder", Author: "jvde-github", version 3
Fetch Web Libraries locally at /etc/AIS-catcher/webassets
Detached kernel driver
Found Rafael Micro R828D tuner
RTL-SDR Blog V4 Detected
TCP feed: open socket for host: aiscatcher.org, port: 4242, filter: OFF, PERSIST: ON, KEEP_ALIVE: OFF, UUID: 7503372d-b20a-48bf-9182-e8a8173d02f4, JSON: ON, status: pending
Server: reading statistics from /etc/AIS-catcher/stat.bin
TCP Server: start thread at port 8100
HTML Server running at port 8100
Server: starting backup service every 10 minutes.
Device    : Blog V4
Settings  : rate 1536K format CU8 tuner AUTO rtlagc ON biastee OFF buffer_count 24
Model #0 -> (Src: 1, Grp: 1): [AIS engine v0.61] ps_ema ON afc_wide ON droop ON fp_ds OFF dsk OFF
[AIS engine v0.61 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0 msg/s

Should I try a fresh install and not run "sudo apt install rtl-sdr"?

**Reply by jvde-github** [reply]
It is not a driver issue... I see:  ***RTL-SDR Blog V4 Detected*** which suggested the modified driver is there. 
So, from a software perspective I suspect it is fine. I need to think a bit about this. 

Different devices have different performance due to noise levels and interference. Can you let it run for a few hours to see if it is "no reception at all" or just poorer reception?

**Reply by jjd7076** [reply]
Absolutely.  I'll report back in a few hours.  Thank you for your help with this!

**Reply by jvde-github** [reply]
Are you sure it is not a hardware issue of some sort. Bit puzzled to be honest.... But will give it a bit more thought. Make sure, when you run, you see  "RTL-SDR Blog V4 Detected" in the output.

**Reply by jjd7076** [reply]
Just to follow up, about an hour ago I started receiving messages from a ship nearby.  I'm still receiving messages from It.  I started picking it up when it was about a 4 miles away, So I am getting signals close by.  Before I was receiving about 9 ships up to 15-20 miles away.  I have the antenna about 25' in the air.  I'm getting a LNA in the mail today, so I'm going to try that later once it arrives.  In windows I was able to set the gain to 48 and that seemed optimal.  Any tips on adjusting the gain on the pi?  There's a coast guard base about 10 miles south of me, so I should be picking up several ships atm.

Again, thank you for your help.

**Reply by jvde-github** [reply]
Ok. How do you run the program? As a service or as command line and how do you currently adjust settings? Via the Web GUI?

**Reply by jjd7076** [reply]
It's configured to run as a background service.  I currently adjust the setting via the Web GUI.

**Reply by jvde-github** [reply]
Ok, cool then you can just set the tuner gain higher in the RTL-SDR settings.

**Reply by jjd7076** [reply]
I set the tuner drop down to 48 and the AGC off like I had it in windows.  I'm still only picking up that same ship.  I'll let it run for a while and see what happens.

**Reply by jvde-github** [reply]
Perhaps linked to an issue resolved recently where some parameters where not properly set if there are two input modes used? Anyway, I will close this unless anything to investigate further?
